libraries/exoplayer/src/main/java/androidx/media3/exoplayer/analytics/DefaultAnalyticsCollector.java

```diff
@@ -406,10 +406,24 @@ public final void onLoadStarted(
     EventTime eventTime = generateMediaPeriodEventTime(windowIndex, mediaPeriodId);
     sendEvent(
         eventTime,
-        AnalyticsListener.EVENT_LOAD_STARTED,
+        C.INDEX_UNSET,
```

Comment:
I know we discussed this, and recommended this approach, but thinking about this again, this causes a problem if:
1. An app has a custom `MediaSource` which won't be updated to trigger the new overload
2. The same app registers an `AnalyticsListener` implementation and waits for `onEvents` to be called with `EVENT_LOAD_STARTED`

Because this implementation will no longer be triggered after this CL.

Discussing with @tonihei we're leaning towards making this a breaking change on the `MediaSourceEventListener` interface to avoid all these pitfalls of trying to keep everything working without breaking compilation. I think we should keep both the old and new methods on `AnalyticsListener`, as that is likely the part that most apps are overriding (and its easy to keep the compat behaviour at that level).

I'm going to play around with that approach and update this PR (I'll also fix the test thing i commented on) - so you don't need to take any action right now.